File: evasion/evasion-architecture-master/players/randomClient.py
import socket
import utils
import argparse
import random

HOST = '127.0.0.1'
# dedicated ports for hunter, prey

class Client:
    def __init__(self, name, port, host=HOST):
        self.name = name
        self.host = host
        self.port = port
        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.sock.connect((host, port))

    def playHunter(self):
        """
        Insert the hunter move logic here
        returns [wallTypeToAdd, wallsToDelete]
        wallTypeToAdd is:
            0 for no wall
            1 for horizontal wall
            2 for vertical wall
            3 for diagnoal wall
            4 for counter diagonal wall
        wallsToDelete is a list of indices(corresponding to self.gameState['walls']) of walls to delete. Can be []
        """
        wallsToDel = []
        for i in range(self.gameState['numWalls']):
            toDel = random.randint(0,30)
            if toDel == 0:
                wallsToDel.append(i)
        # randomly build wall if possible
        wallType = random.randint(0,4)
        if self.gameState['maxWalls'] <= self.gameState['numWalls'] - len(wallsToDel):
            wallType = 0
        return [wallType, wallsToDel]

    def playPrey(self):
        """
        Insert the prey move logic here
        return pair (moveX, moveY)
        """
        x = random.randint(-1,1)
        y = random.randint(-1,1)
        return (x,y)

    def playGame(self):
        stream = ""
        while True:
            stream, line = utils.recv(self.sock, stream)
            toSend = None
            if line == "done":
                break
            elif line == "hunter":
                self.hunter = True
            elif line == "prey":
                self.hunter = False
            elif line == "sendname":
                toSend = self.name + str(self.port)
            else:
                self.gameState = utils.stringToGame(line)
                if self.hunter:
                    move = self.playHunter()
                    toSend = utils.parseHunterMove(self.gameState, move)
                else:
                    move = self.playPrey()
                    toSend = utils.parsePreyMove(self.gameState, move)
            if toSend is not None:
                print "sending: " + toSend
                self.sock.sendall(toSend + "\n")

if __name__== "__main__":
    parser = argparse.ArgumentParser(description='')
    parser.add_argument('--port', type=int, default= 9000, help='port')
    parser.add_argument('--name', type=str, default= "randomPlayer")
    args = parser.parse_args()

    port = args.port
    name = args.name

    player = Client(name, port)
    player.playGame()

File: evasion/evasion-architecture-master/src/main/java/edu/nyu/cs/hps/evasion/RandomClient.java
package edu.nyu.cs.hps.evasion;

import edu.nyu.cs.hps.evasion.EvasionClient;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class RandomClient extends EvasionClient {

    RandomClient(String name, int port) throws IOException {
        super(name, port);
    }

    public HunterMove playHunter() throws Exception {
        List<Integer> wallsToDel = new ArrayList<>();
        Random random = new Random();
        for (int i = 0; i < this.gameState.walls.size(); i++) {
            int toDel = random.nextInt(31);
            if (toDel == 0) {
                wallsToDel.add(i);
            }
        }

        int wallType = random.nextInt(5);
        if (this.gameState.maxWalls <= this.gameState.walls.size() - wallsToDel.size()) {
            wallType = 0;
        }
        return new HunterMove(wallType, wallsToDel);
    }

    public PreyMove playPrey() {
        Random random = new Random();
        return new PreyMove(random.nextInt(3) - 1, random.nextInt(3) - 1);
    }

    public static void main(String[] args) throws Exception {
        if (args.length != 2) {
            System.out.println("Usage: java RandomClient <port> <name>");
        }
        int port = Integer.parseInt(args[0]);
        String name = args[1];
        EvasionClient evasionClient = new RandomClient(name, port);
        evasionClient.playGame();
        evasionClient.socket.close_socket();
    }

}
